File: apps/android/feature/scribble/domain/src/main/java/com/scribblefit/feature/scribble/domain/ScribbleRepository.kt
package com.scribblefit.feature.scribble.domain

import com.scribblefit.core.model.Scribble
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for scribbles.
 */
interface ScribbleRepository {
    suspend fun insertScribble(scribble: Scribble): Long
    suspend fun updateScribble(scribble: Scribble)
    suspend fun deleteScribble(scribbleId: Long)
    fun getScribble(scribbleId: Long): Flow<Scribble>
    fun getScribblesByDate(date: Long): Flow<List<Scribble>>
}

File: apps/ios/ScribbleFit/Sources/Domain/Repositories/ScribbleRepository.swift
import Foundation

@MainActor
public protocol ScribbleRepository: Sendable {
    func getScribbles(for date: Date) -> AsyncStream<[Scribble]>
    func addScribble(_ scribble: Scribble) async throws
    func updateScribble(_ scribble: Scribble) async throws
    func deleteScribble(id: UUID) async throws
    func getScribble(id: UUID) async throws -> Scribble?
}
